app/v1/models/caterers.py

```diff
@@ -0,0 +1,19 @@
+from app.v1.models.users import DbUsers
+
+
+class DbCaterers(DbUsers):
+    def __init__(self):
+        super().__init__()
+
+    def add_user(self, email, username, password, address, category='caterer', brand_name=''):
+        if email not in self.all_emails:
+            try:
+                self.all_users[username]
+            except KeyError:
+                brand_name_final = brand_name if brand_name else username
```

Comment:
I thought we had talked about this. Except/Catch only report errors, they don't implement actual code. And I dont see validation being done here.